### PR TITLE
Remove etc/pyenv.d/rehash/envs.bash

### PR DESCRIPTION
**Pull request by blueyed**
There is no need for pyenv-virtualenv to rehash all versions again,
since pyenv takes care of this already.

I have compared the output from before and after, and it is the same.

For me it improves `pyenv rehash` from 128ms to 89ms.

Fixes https://github.com/pyenv/pyenv-virtualenv/issues/274.
Ref: https://github.com/pyenv/pyenv/issues/784#issuecomment-404876430

Used `python -m timeit "__import__('subprocess').Popen(['pyenv', 'rehash']).communicate()"`.

Might be also interesting to know how it would interact with https://github.com/pyenv/pyenv/pull/1185 in general, which is meant to only rehash the current version - and this would still add all (virtualenv) shims into every separate shim dir.

### CODE REVIEW

File: etc/pyenv.d/rehash/envs.bash [hunk 1]
```diff
@@ -1,11 +0,0 @@
-virtualenv_list_executable_names() {
-  local file
-  shopt -s nullglob
-  for file in "$PYENV_ROOT"/versions/*/envs/*/bin/*; do
```
Comment (yyuu) [inline etc/pyenv.d/rehash/envs.bash]: IIRC pyenv's rehash doesn't scan "envs" directory, isn't it? Without this, shims for virtualenv executables won't be creatwd, I think 🤔

Comment (blueyed) [reply etc/pyenv.d/rehash/envs.bash]: I tested it, but might have missed something, or might not have a required setup.
But I am currently not sure about if and where `envs` are used.

Comment (blueyed) [reply etc/pyenv.d/rehash/envs.bash]: This is quite a blocker for https://github.com/pyenv/pyenv/pull/1185 - which works quite well apart from that already.

**Review by native-api — APPROVED** [review]
Atm, all `versions/*/envs/*` are also symlinked from `versions/`, that's how Pyenv takes care of them.

I'm not sure if this happens in all cases though. @blueyed , could you check the code to verify that?

**Comment by native-api** [comment]
Judging by https://github.com/pyenv/pyenv-virtualenv/blob/3c0edf888ef8798bec2824f3f1a721d879bada3f/bin/pyenv-virtualenv#L584-L587, symlinks in `versions/` were intended to be a temporary solution "for backward compatibility" (not sure compatibility with what that's talking about tbh).

So we cannot delete this.

We can however adapt it to skip any environments that are symlinked to from `versions/`.

**Comment by native-api** [comment]
> We can however adapt it to skip any environments that are symlinked to from `versions/`.

And do that early so it's as close to a no-op as possible when everything is symlinked.

**Pull request closed without merging**